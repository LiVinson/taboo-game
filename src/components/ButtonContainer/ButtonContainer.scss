.btnContainer {
    margin-top:1.5rem;
    display:flex;
    justify-content: space-between;
    
}