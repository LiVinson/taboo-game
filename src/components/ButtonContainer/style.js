import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`