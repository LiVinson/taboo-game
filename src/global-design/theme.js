const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
