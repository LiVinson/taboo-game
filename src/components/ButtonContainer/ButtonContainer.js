import React from "react"
import PropTypes from "prop-types"
import { Button, PrimaryButton, LargeButton } from "components/shared/Button"
import { StyledButtonContainer } from "./style"

//Appears at bottom of Taboo Card when a buttons prop is passed to taboo.
// Receives array of objects with button text and click handler
const ButtonContainer = ({ buttons }) => {
  return (
    <StyledButtonContainer>
      {buttons.map((buttonInfo, index, buttons) =>
        determineButtonType(buttonInfo, index, buttons.length)
      )}
    </StyledButtonContainer>
  )
}

ButtonContainer.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      onClick: PropTypes.func.IsRequired,
      type: PropTypes.string,
    })
  ),
}

/*3 possible buttons: regular, primary, and large regular
  When there are 2 or more buttons, first is always button (white) and second is primary button (color).
  If there are less than 2 buttons or more than 2 (e.g. 1 or 3 buttons) then that button is large button
  Margin is only added to large button when there are multiple buttons to separate it from buttons above
*/
const determineButtonType = (buttonInfo, index, length) => {

  if (index === 0 && length >= 2) {
    return <Button key={index} {...buttonInfo} />
  } else if (index === 1 && length >= 2) {
    return <PrimaryButton key={index} {...buttonInfo} />
  } else {
    return <LargeButton margin={length>2} key={index} {...buttonInfo} />
  }
}

export default ButtonContainer