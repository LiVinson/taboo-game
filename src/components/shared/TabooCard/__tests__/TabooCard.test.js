import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import {
	TabooCard,
	ButtonTabooCard,
	LayeredTabooCard,
	TabooCardTop,
	TabooWordContainer,
	TabooList,
	TabooListItem,
	TabooBody,
	FilteredTabooList,
} from '../TabooCard'
import ButtonContainer from 'components/ButtonContainer'

describe('TabooCard renders and functions correctly', () => {
	const defaultProps = {
		tabooWord: 'taboo',
		list: ['word1', 'word2', 'word3'],
	}

	// ------------------- Main Taboo Card--------------------
	test('TabooCard renders correctly when list is passed ', () => {
		const props = {
			...defaultProps,
		}
		// requires TP due to nested components using theme prop
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<TabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('TabooCard renders children prop', () => {
		const h1 = <h1>I am a child property</h1>
		const props = {
			...defaultProps,
			list: null,
			children: h1,
		}

		const wrapper = shallow(<TabooCard {...props} />)
		expect(wrapper.contains(<h1>I am a child property</h1>)).toEqual(true)
	})
})
// ------------------- Button Taboo Card--------------------

describe('ButtonTabooCard renders correctly', () => {
	const defaultProps = {
		tabooWord: 'taboo',
		list: ['word1', 'word2', 'word3'],
		buttons: [
			{ text: 'Back', onClick: jest.fn() },
			{ text: 'Next', onClick: jest.fn() },
		],
	}

	test('ButtonTabooCard renders ButtonContainer', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<ButtonTabooCard {...props} />)
		expect(wrapper.find(ButtonContainer)).toHaveLength(1)
	})

	test('ButtonTabooCard renders correctly when list is provided', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<ButtonTabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('ButtonTabooCard renders children prop if provided', () => {
		const h1 = <h1>I am a child property</h1>
		const props = {
			...defaultProps,
			list: null,
			children: h1,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<ButtonTabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})
})

// ------------------- Layered Taboo Card--------------------

describe('Layered TabooCard renders correctly', () => {
	const defaultProps = {
		tabooWord: 'taboo',
		list: ['Taboo!', 'Taboo!', 'Taboo!', 'Taboo!', 'Taboo!'],
		layer: 'top',
	}

	test('Top LayeredTabooCard renders correctly', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<LayeredTabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()

		expect(wrapper).toHaveStyleRule('position', 'absolute')
		expect(wrapper).toHaveStyleRule('z-index', '-5')
		expect(wrapper).toHaveStyleRule('transform', 'translate(-50%,-50%) rotate(-20deg)')
		expect(wrapper).toHaveStyleRule('color', theme.color.accent2)
	})

	test('Middle LayeredTabooCard renders correctly', () => {
		const props = {
			...defaultProps,
			layer: 'middle',
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<LayeredTabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
		expect(wrapper).toHaveStyleRule('position', 'absolute')
		expect(wrapper).toHaveStyleRule('z-index', '-10')
		expect(wrapper).toHaveStyleRule('transform', 'translate(-50%,-50%) rotate(16deg)')
		expect(wrapper).toHaveStyleRule('color', theme.color.accent1)
	})

	test('Bottom LayeredTabooCard has correct styles', () => {
		const props = {
			...defaultProps,
			layer: 'bottom',
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<LayeredTabooCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
		expect(wrapper).toHaveStyleRule('position', 'absolute')
		expect(wrapper).toHaveStyleRule('z-index', '-15')
		expect(wrapper).toHaveStyleRule('transform', 'translate(-50%,-50%) rotate(-5deg)')
		expect(wrapper).toHaveStyleRule('color', theme.color.accent3)
	})
})

// ------------------- Taboo Card Top --------------------

test('TabooCardTop renders correctly', () => {
	const props = {
		tabooWord: 'Taboo!',
	}
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<TabooCardTop {...props} />
			</ThemeProvider>
		)
		.toJSON()
	expect(wrapper).toMatchSnapshot()
})

// ------------------- FilteredTabooList --------------------
describe('FilteredTabooList renders correctly', () => {
	const unfilteredList = [
		{ name: 'Player 1', team: 'team 1', playerId: '12345' },
		{ name: 'Player 2', team: 'team 1', playerId: '23456' },
		{ name: 'Player 3', team: 'unassigned', playerId: '34567' },
		{ name: 'Player 4', team: 'unassigned', playerId: '45678' },
	]
	let filterValue = 'team 1'
	const noneMessage = `No ${filterValue} players`

	const defaultProps = {
		unfilteredList,
		filterKey: 'team',
		filterValue,
		displayProperty: 'name',
		listTitle: 'team 1',
		noneMessage,
	}

	test('FilterTabooList only lists values based on filterKey and filterValue', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<FilteredTabooList {...props} />
			</ThemeProvider>
		)
		
		expect(wrapper.find(TabooListItem).length).toBe(2)
		expect(wrapper.find(TabooListItem).at(0).text()).toEqual("Player 1")
		expect(wrapper.find(TabooListItem).at(1).text()).toEqual("Player 2")

	})

	test('FilterTabooList displays none message if no values for filterKey and filterValue in list', () => {
		const props = {
			...defaultProps,
			filterValue: "team 2"
		}

		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<FilteredTabooList {...props} />
			</ThemeProvider>
		)
		
		expect(wrapper.find(TabooListItem).length).toBe(1)
		expect(wrapper.find(TabooListItem).text()).toEqual(noneMessage)
	})
})
