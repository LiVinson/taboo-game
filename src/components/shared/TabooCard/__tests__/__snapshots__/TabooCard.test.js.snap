// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTabooCard renders correctly ButtonTabooCard renders children prop if provided 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c4:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c4:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c4:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c4:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c5 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c5:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c5:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c5:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c5:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c5:focus {
  outline: solid 2px #f7f9f9;
}

.c3 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <h1>
      I am a child property
    </h1>
  </div>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[MockFunction]}
    >
      Back
    </button>
    <button
      className="c5"
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`ButtonTabooCard renders correctly ButtonTabooCard renders correctly when list is provided 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

.c6 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c6:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c6:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c6:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c6:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c7 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c7:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c7:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c7:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c7:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c7:focus {
  outline: solid 2px #f7f9f9;
}

.c5 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        word1
      </li>
      <li
        className="c4"
      >
        word2
      </li>
      <li
        className="c4"
      >
        word3
      </li>
    </ul>
  </div>
  <div
    className="c5"
  >
    <button
      className="c6"
      onClick={[MockFunction]}
    >
      Back
    </button>
    <button
      className="c7"
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Layered TabooCard renders correctly Bottom LayeredTabooCard has correct styles 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
  position: absolute;
  top: 66%;
  left: 50%;
  margin-bottom: 0rem;
  color: #a23b72;
  z-index: -15;
  -webkit-transform: translate(-50%,-50%) rotate(-5deg);
  -ms-transform: translate(-50%,-50%) rotate(-5deg);
  transform: translate(-50%,-50%) rotate(-5deg);
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c0:before {
  display: none;
}

.c0:after {
  display: inline-block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: lightgray;
  opacity: 0.5;
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
    </ul>
  </div>
</div>
`;

exports[`Layered TabooCard renders correctly Middle LayeredTabooCard renders correctly 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
  position: absolute;
  top: 66%;
  left: 50%;
  margin-bottom: 0rem;
  color: #8AC926;
  z-index: -10;
  -webkit-transform: translate(-50%,-50%) rotate(16deg);
  -ms-transform: translate(-50%,-50%) rotate(16deg);
  transform: translate(-50%,-50%) rotate(16deg);
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c0:before {
  display: none;
}

.c0:after {
  display: inline-block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: lightgray;
  opacity: 0.5;
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
    </ul>
  </div>
</div>
`;

exports[`Layered TabooCard renders correctly Top LayeredTabooCard renders correctly 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
  position: absolute;
  top: 66%;
  left: 50%;
  margin-bottom: 0rem;
  color: #ee5b2c;
  z-index: -5;
  -webkit-transform: translate(-50%,-50%) rotate(-20deg);
  -ms-transform: translate(-50%,-50%) rotate(-20deg);
  transform: translate(-50%,-50%) rotate(-20deg);
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c0:before {
  display: none;
}

.c0:after {
  display: inline-block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: lightgray;
  opacity: 0.5;
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
      <li
        className="c4"
      >
        Taboo!
      </li>
    </ul>
  </div>
</div>
`;

exports[`TabooCard renders and functions correctly TabooCard renders correctly when list is passed  1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c4 {
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  position: relative;
  padding-bottom: 1.5rem;
}

.c3 {
  list-style-type: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    taboo
  </div>
  <div
    className="c2"
  >
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        word1
      </li>
      <li
        className="c4"
      >
        word2
      </li>
      <li
        className="c4"
      >
        word3
      </li>
    </ul>
  </div>
</div>
`;

exports[`TabooCardTop renders correctly 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
